drivers/at/at.c

```diff
@@ -88,18 +96,48 @@ static void _isrpipe_write_one_wrapper(void *_dev, uint8_t data)
 #endif
 }
 
-int at_dev_init(at_dev_t *dev, uart_t uart, uint32_t baudrate, char *buf, size_t bufsize)
+int at_dev_init(at_dev_t *dev, at_dev_init_t const *init)
 {
-    dev->uart = uart;
+    dev->uart = init->uart;
+    dev->err_buf = init->err_buf;
+    dev->err_buf_size = init->err_buf_size;
 
 #if IS_USED(MODULE_AT_URC_ISR)
     dev->awaiting_response = false;
     dev->event.handler = _event_process_urc;
 #endif
 
-    isrpipe_init(&dev->isrpipe, (uint8_t *)buf, bufsize);
+    isrpipe_init(&dev->isrpipe, (uint8_t *)init->rx_buf, init->rx_buf_size);
+
+    return uart_init(init->uart, init->baudrate, _isrpipe_write_one_wrapper, dev);
+}
 
-    return uart_init(uart, baudrate, _isrpipe_write_one_wrapper, dev);
+int at_parse_resp(at_dev_t *dev, char const *resp)
+{
+    if (*resp == '\0') {
+        return 1;
+    }
+    if (strncmp(CONFIG_AT_RECV_OK, resp, strlen(CONFIG_AT_RECV_OK)) == 0) {
+        dev->err_buf[0] = '\0';
+        return 0;
+    }
+    if (strncmp(CONFIG_AT_RECV_ERROR, resp, strlen(CONFIG_AT_RECV_ERROR)) == 0) {
+        return -1;
+    }
+    size_t const cmx_prefix_len = strlen("+CMx ERROR: ");
+    if (strncmp("+CME ERROR: ", resp, cmx_prefix_len) &&
+        strncmp("+CMS ERROR: ", resp, cmx_prefix_len)) {
+        // A command may return either one or the other, we need not differentiate
```

Comment:
The comment should mention that it is the "otherwise" condition.

In other words, it is NOT prefix `"+CME ERROR: "` AND it is NOT prefix `"+CMS ERROR: "`.